lib/android/src/main/java/com/airbnb/android/react/maps/AirMapWMSTile.java

```diff
@@ -50,6 +62,74 @@ public synchronized URL getTileUrl(int x, int y, int zoom) {
       return url;
     }
 
+    class AIRMapUrlTile implements TileProvider {
+      private static final int BUFFER_SIZE = 4 * 1024;
+      private int width;
+      private int height;
+      private ReadableMap requestProperties;
+
+      AIRMapUrlTile(int width, int height, ReadableMap requestProperties) {
+        this.width = width;
+        this.height = height;
+        this.requestProperties = requestProperties;
+      }
+
+      private byte[] readTileImage(int x, int y, int zoom) {
+        InputStream in = null;
+        ByteArrayOutputStream buffer = null;
+        URL url = getTileUrl(x, y, zoom);
+        if (url == null)
+          return null;
+
+        try {
+          if(getUrlTemplate().contains("https")) {
+            HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+            for (Map.Entry<String, Object> entry : requestProperties.toHashMap().entrySet()) {
+              connection.addRequestProperty(entry.getKey(), (String) entry.getValue());
+            }
+            connection.connect();
+            in = connection.getInputStream();
```

Comment:
this can be abstracted to private fnc

Reply:
d8b9a563c2d6e345fd2b9c6430bfc91eb3831e5f